awpy/analytics/map_control.py

```diff
@@ -0,0 +1,413 @@
+"""Functions for calculating map control values and metrics.
+
+    A team's map control can be thought of as the sum of it's
+    individual player's control.
+
+    Example notebook:
+
+    # pylint: disable=line-too-long
+    github.com/pnxenopoulos/awpy/blob/main/examples/05_Map_Control_Calculations_And_Visualizations.ipynb
+"""
+
+from collections import defaultdict, deque
+
+import numpy as np
+
+from awpy.analytics.nav import area_distance, calculate_tile_area, find_closest_area
+from awpy.data import NAV, NAV_GRAPHS
+from awpy.types import (
+    BFSTileData,
+    FrameMapControlValues,
+    FrameTeamMetadata,
+    GameFrame,
+    GameRound,
+    PlayerPosition,
+    TeamFrameInfo,
+    TeamMapControlValues,
+    TeamMetadata,
+    TileDistanceObject,
+    TileId,
+    TileNeighbors,
+)
+
+
+def _approximate_neighbors(
+    map_name: str,
+    source_tile_id: TileId,
+    n_neighbors: int = 5,
```

Comment:
do you think this default is globally ok? are there some maps were it might not work well?

Reply:
Hm I'm not sure exactly but there could be a way to find out. So this function is run for navigation tiles that don't have navigable tile neighbors. So then I find the `n_neighbors` closest tiles to the target tile to approximate "neighbors" so the bfs algorithm can run as expected and not ignore players in these isolated tiles. So in my head I thought 5 would be a good default value. Do you know what would be a good way to estimate a good defualt value or should we be leave it to be set by the user with no default? Another idea I had was to make it a max_distance parameter instead of n_neighbors and try to estimate a good distance metric based on isolated tiles in maps but could be overdoing it lol

Reply:
Got it, I think it's fine for now. I would just test in a few cases to see how sensitive it is. Mostly vibe-checking the parameter across some maps